src/nisarqa/input_product_readers/insar_product.py

```diff
@@ -203,5 +204,53 @@ def save_qa_metadata_to_h5(self, stats_h5: h5py.File) -> None:
                 ds_description=f"Polarizations for Frequency {freq}.",
             )
 
+    def _path_to_orbit_grp(self) -> str:
+        """Path in HDF5 to the product's orbit metadata group."""
+        pass
+
+    def get_orbit(self, ref_or_sec: str) -> isce3.core.Orbit:
+        """
+        Get the isce3.core.Orbit for the input granule.
+
+        Parameters
+        ----------
+        ref_or_sec : str
+            One of {"reference", "secondary"}.
+            InSAR granules are generated from two input RSLC granules,
+            referred to as the "reference RSLC" and the "secondary RSLC".
+            Specifiying "reference" will return the Orbit for the
+            reference RSLC, and  specifying "secondary" will
+            return the Orbit for the secondary RSLC.
+
+        Returns
+        -------
+        orbit : isce3.core.Orbit
+            The Orbit for the requested reference or secondary RSLC.
+        """
+
+        if ref_or_sec not in ("reference", "secondary"):
+            raise ValueError(
+                f"{ref_or_sec=}, must be either 'reference' or 'secondary'."
+            )
+
+        orbit_path = "/".join([self._metadata_group_path, "orbit", ref_or_sec])
+        with h5py.File(self.filepath, "r") as f:
+            try:
+                orbit_grp = f[orbit_path]
+            except KeyError as e:
+                if nisarqa.Version.from_string(
+                    self.product_spec_version
+                ) < nisarqa.Version.from_string("1.1.1"):
+                    # "reference" and "secondary" orbit sub groups were
+                    # added in product specs 1.1.1
+                    orbit_path = "/".join([self._metadata_group_path, "orbit"])
+                    orbit_grp = f[orbit_path]
```

Comment:
Is this the reference orbit? Should we raise an exception if the user asked for the "secondary" orbit?